src/core/swap.c

```diff
@@ -256,6 +256,9 @@ static int swap_verify(Swap *s) {
         if (!unit_has_name(UNIT(s), e))
                 return log_unit_error_errno(UNIT(s), SYNTHETIC_ERRNO(ENOEXEC), "Value of What= and unit name do not match, not loading.");
 
+        if (exec_needs_pid_namespace(&s->exec_context))
```

Comment:
Hmm, actually this can be further dedupped by checking unit_vtable->notify_pidref in unit_verify_contexts I think. But feel free to ignore.